source/1_engine/rex_engine/src/string/stringid.cpp

```diff
@@ -0,0 +1,211 @@
+  }
+  //-------------------------------------------------------------------------
+  bool StringID::operator!=(const SID& name) const
+  {
+    return m_comparison_index != static_cast<uint32>(name);
+  }
+  //-------------------------------------------------------------------------
+  bool StringID::operator==(const StringEntryID& entryID) const
+  {
+    return m_comparison_index == static_cast<uint32>(entryID);
+  }
+  //-------------------------------------------------------------------------
+  bool StringID::operator!=(const StringEntryID& entryID) const
+  {
+    return m_comparison_index != static_cast<uint32>(entryID);
+  }
+
+  //-------------------------------------------------------------------------
+  /** True for StringID(), StringID(NAME_None) and StringID("None") */
+  bool StringID::is_none() const
+  {
+    return *this == SID::None;
+  }
+
+  //-------------------------------------------------------------------------
+  StringID create_sid(const SID& name)
+  {
+    return StringID(name);
+  }
+  //-------------------------------------------------------------------------
+  StringID create_sid(const char* characters)
+  {
+    return StringID(characters, rsl::strlen(characters));
+  }
+  //-------------------------------------------------------------------------
+  StringID create_sid(const char* characters, s32 size)
+  {
+    return StringID(characters, size);
+  }
+  //-------------------------------------------------------------------------
+  StringID create_sid(const rsl::string& string)
+  {
+    return StringID(string.data(), string.size());
+  }
+  //-------------------------------------------------------------------------
+  StringID create_sid(const rsl::string_view& stringView)
+  {
+    return StringID(stringView.data(), stringView.size());
+  }
+
+  //-------------------------------------------------------------------------
+  bool operator==(const rsl::string& s, const StringID& sid)
+  {
+    return create_sid(s) == sid;
+  }
+  //-------------------------------------------------------------------------
+  bool operator!=(const rsl::string& s, const StringID& sid)
+  {
+    return create_sid(s) != sid;
+  }
+  //-------------------------------------------------------------------------
+  bool operator==(const StringID& sid, const rsl::string& s)
+  {
+    return s == sid;
+  }
+  //-------------------------------------------------------------------------
+  bool operator!=(const StringID& sid, const rsl::string& s)
+  {
+    return s != sid;
+  }
+} // namespace rex
+
+//-------------------------------------------------------------------------
+rex::StringID operator""_sid(const char* string, size_t size)
+{
+  return rex::StringID(string, static_cast<u32>(size)); // NOLINT(cppcoreguidelines-narrowing-conversions)
+}
+
+//-------------------------------------------------------------------------
+rsl::ostream& operator<<(rsl::ostream& os, const rex::StringID& stringID)
+{
+  os << stringID.to_string();
```

Comment:
one of the reasons why this should return a view.
otherwise you have a string allocation every time you call this which is exactly what you want to avoid with string pooling